### PR TITLE
State memory filter

### LINKED ISSUE #506: Support State Memory for Filter Collapsing
During Q4, we are introducing additional filters to CC Search. We know we will be adding even more filters in the future, both for images and other content types. We are also changing how filters are presented, to be listed in a left hand side bar. [See issue 529] (https://github.com/creativecommons/cccatalog-frontend/issues/529).

At present, we are building the left hand side bar in such a way that filters are shown (i.e. the section is expanded) when the page is loaded. This is to ensure that users are aware of filters. However, as we expand to more filters being included in the side bar, we anticipate this becoming a frustrating user experience, especially for power users, who have filtered and collapsed/minimized the filter section on the results page.

## Proposed Implementation
We'd like to support session-based state memory for filter expansion/collapsing, meaning we'd remember when a user has chosen to expand or collapse the left hand filter side bar. When users reload or go back to the results page, the filter side bar will be in the state that they left them. The default for new sessions will remain expanded for now.

### PR DESCRIPTION
**Pull request by SomtochiAma**
Fixes #506

[Short description explaining the high-level reason for the pull request]
Storing of filters in localStorage so that users can still have the same filters when they reload the page
 
Developer Certificate of Origin Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors. 1 Letterman Drive Suite D4700 San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

### CODE REVIEW

File: src/components/SearchGridFilter.vue [hunk 1]
```diff
@@ -135,10 +139,12 @@ export default {
   },
   methods: {
     onUpdateFilter() {
+      console.log('here');
```
Comment (brenoferreira) [inline src/components/SearchGridFilter.vue]: Please remove this `console.log` call

File: src/components/SearchGridFilter.vue [hunk 2]
```diff
@@ -147,6 +153,7 @@ export default {
       Object.keys(this.filter).forEach((key) => {
         filter[key] = transformFilterValue(this.filter, key);
       });
+      localStorage.setItem('filter', this.filter);
```
Comment (brenoferreira) [inline src/components/SearchGridFilter.vue]: Calls to `localStorage` methods should be in a store action or mutation, not inside components.

**Comment by brenoferreira** [comment]
@SomtochiAma we're making some design changes to the filters (seen in #533). We're still not sure if we're going to go forward with this. I'll keep this Pull Request open in the meantime in case we decide we're going to have this feature in the product.

**Comment by brenoferreira** [comment]
@SomtochiAma we're going ahead with this. If you want to fix the merge conflicts, we'll review this again and if all looks good, we'll merge

**Comment by SomtochiAma** [comment]
Sure. I will get it done.

**Comment by akmadian** [comment]
@SomtochiAma Are you still working on this? If we do not see some more progress on this issue in the next two days, we will close this PR and free the issue up for others to work on.

**Comment by SomtochiAma** [comment]
> @SomtochiAma Are you still working on this? If we do not see some more progress on this issue in the next two days, we will close this PR and free the issue up for others to work on.

I am not sure I will get it done. Let me close the PR.

**Pull request closed without merging**